src/main/java/InfrastructureManager/Master.java

```diff
@@ -1,37 +1,138 @@
 package InfrastructureManager;
 
+import java.util.ArrayList;
+
+/**
+ * Master Class of the Infrastructure Manager, singleton class
+ */
 public class Master {
 
     private final CommandSet commandSet;
-    private final MasterInput input;
-    private final MasterOutput output;
+    private final ArrayList<Runner> runnerList;
+    private final ArrayList<Runner> running; //List to store currently running runners
```

Comment:
I am not a fan of  using 2 lists, someone is going to start a thread and forget to add it to the other list.
There should be no problem in using 1 list, but if there is then use a small class for it.

Reply:
You are right, I've already solved it by getting rid of the `runnning` list, and getting that info direclty from the Runner. I will resolve this comment